drivers/i2c/i2c_ll_stm32_v2.c

```diff
@@ -1302,5 +1322,26 @@ int i2c_stm32_transaction(const struct device *dev,
 		msg.len = rest;
 	} while (rest > 0U);
 
+#ifndef CONFIG_I2C_STM32_INTERRUPT
+	struct i2c_stm32_data *data = dev->data;
```

Comment:
Hi @erwango @GeorgeCGV @etienne-lms, two variables are missing here:
```
	struct i2c_stm32_data *data = dev->data;
	const struct i2c_stm32_config *cfg = dev->config;
	I2C_TypeDef *i2c = cfg->i2c;
```
What is the procedure to follow?

Reply:
Indeed we missed that :-\
How come CI build test did not found this issue.

Can you create a pull request to get this fixed?